scalyr_agent/configuration.py

```diff
@@ -359,23 +362,39 @@ def parse(self):
                     "serverHost"
                 ] = self.__get_default_hostname()
 
-            # Add in implicit entry to collect the log generated by this agent.
+            # Add in implicit entry to collect the logs generated by this agent and its worker sessions.
             agent_log = None
+            worker_session_agent_logs = None
             if self.implicit_agent_log_collection:
                 # set path as glob to handle log files from the multiprocess worker sessions.
-                config = JsonObject(
-                    path="agent*.log",
-                    exclude=JsonArray("*_debug.log"),
+                config = JsonObject(path="agent.log", parser="scalyrAgentLog",)
+
+                # add log config for the worker session agent log files.
+                worker_session_logs_config = JsonObject(
```

Comment:
I assume we want to add this one even if multiple sessions are not configured, right?

(in case user changes the configuration, to make sure old / previous logs are still picked up)

Reply:
Correct. We create a new log matcher for the worker logs anyway.